### DISCUSSION TITLE
Revisit nmdc submission portal sandbox deletion policy

### DISCUSSION THREAD

**Discussion by mslarae13** [microbiomedata/issues#743] [General]
Currently, the NMDC submission portal sandbox is wiped every sunday.

Some ambassadors or other users are still using the production environment for tests, workshops, and fake submissions.

@ljohnson09 and @jkelliher-github made the valid point that some people might want their test submission to stick around for longer.
- Prep for a workshop on Friday, and the submission still be there for monday's session
- Re-use the same submission for multiple workshops & not need to re-submit

I propose a couple solution options here, would like thoughts / feedback before creating an issue for targeted resolution

**Solution 1**
Provide a "delete this submission" button. So the sandbox NEVER gets wiped. People can just delete their submissions when they're read 

**Solution 2**
Extend the time. 
- Submissions are deleted ever quarter
- Submissions are deleted 90 days after creation
  - Should provide a warning? How?
- Add a "trash" where people can restore the submission for so many days after deletion

**Solution 3**
Add a "delete this is a test" check box to the production site & those get deleted every so many days
- This checkbox can be check after the fact so it's not deleted until someone says they want it to be. 
  - I think a LOT of people make test submission and forget about them, so I would like the ability to delete them & send a notice

Any other suggestions?

**Comment by mslarae13** [comment]
@pkalita-lbl 
@naglepuff 
@lamccue 
@jkelliher-github 
@ljohnson09 
@frodriguez16 
@simroux 

All your feedback / thoughts would be appreciated!

**Comment by lamccue** [comment]
For the Sandbox, could we consider a combination of Solution 1 and Solution 2:

- Provide a "delete this submission" button. (We can encourage users to keep their submissions cleaned up, but let's be realistic, they won't)
- _And_ all submissions will be deleted twice a year, say June 30 and December 31.  (doesn't seem like a more frequent clean up is necessary?)

I'm not keen on the idea of implementing a "trash"

For the Production site, maybe there is another option:

- On the landing page, maybe next to the "create new submission" button, provide an option to "create a test submission", which would automagically take the user to the Sandbox.

**Comment by simroux** [comment]
I like @lamccue suggestions, purging twice a year and removing any submission older than a year seems reasonable (and should take care of most cases mentioned), along with an easy way for users to delete their submission when they need to.

For the link from the main submission portal, I don't know if we want the sandbox to be "that" open, but if we do, maybe that's a fourth link next to "Quickstart" / "Tutorial" / "Reference", e.g. "Testing site" (or a better way of wording this ?)

**Comment by pkalita-lbl** [comment]
Just from a technical perspective, I would be hesitant to go for anything that makes the site itself behave differently on production vs sandbox i.e. sandbox has this button/link and production doesn't. Clearing the database periodically is not a feature of the site itself, so that's fine. It's not impossible to make the site behave differently in different environments, but it just makes things harder to maintain and test.

To me, adding a UI affordance (not dependent on environment) to delete submissions is a no-brainer. I feel like it should be done regardless of this discussion. There's already an API endpoint for it. We'd just need to decide which submission permission roles are allowed to delete and add it to the UI. Then Montana (or any other admin) could delete anything she wanted to, and users could delete their own on sandbox, production, dev, whatever.

As far as changing the deletion policy on sandbox I don't have a strong opinion. I would just point out that currently we clear _all_ submissions once a week. But in addition to changing the frequency we can also consider changing _which_ submissions are selected for deletion. For example, we could, say, once a month delete anything that was created over six months ago. We don't actually currently track the last time a submission was _udpated_ (only created), but that wouldn't be hard to track if that would also be helpful in defining a deletion policy.

**Comment by mslarae13** [comment]
Discussed with @naglepuff , @pkalita-lbl , @JamesTessmer , @lamccue 

**Summary/ Decisions (*=pending discussion with team)**

- Add “delete this submission” button
- Anyone with “creator role” (submitter & PIs) and admins can delete a submission
- *Eliminate sandbox
  - dataportal sandbox is confusing
  - Users forget about / don't access sandbox 
  - One place makes it easier
  - Less code to manage
- Add “Create test submission” button to prod+dev
  - Test, fake, workshop, idk what the word is
- Create/ provide clear tool tip text to elaborate / explain when to make a new submission vs test submission
- Test submissions can NOT be submitted
- Test submissions can NOT be converted to real submissions
  - Test is not equal to Draft
- Eliminate deletion policy until we get a good sense of HOW test submissions are being created & IF people delete their own tests
- Add a “date last modified” attribute to submissions
- Add a “test” option to the app
- Add ability to track what submissions in the submission portal come from the app (for metrics)
- *Update all documentation that references the -sandbox.
  - *Include newsletter notice

**Reply by mslarae13** [reply]
Document in ADR when approved

**Reply by mslarae13** [reply]
ADR draft: https://github.com/microbiomedata/NMDC_documentation/pull/67

**Comment by mslarae13** [comment]
Decisions made. See ADR: https://github.com/microbiomedata/issues/blob/main/decisions/0014-eliminate-sandbox-environment.md
